src/CodeGeneration.Roslyn.Engine/CompilationGenerator.cs

```diff
@@ -31,37 +30,58 @@ public class CompilationGenerator
+        /// <param name="pluginPaths">The paths to plugins.</param>
+        /// <param name="intermediateOutputDirectory">The path to the directory that contains generated source files.</param>
+        /// <param name="buildProperties">The build properties to expose to generators.</param>
+        public CompilationGenerator(string? projectDirectory, IReadOnlyList<string> compile, IReadOnlyList<string> referencePath, IEnumerable<string> preprocessorSymbols, IReadOnlyList<string> pluginPaths, string intermediateOutputDirectory, IReadOnlyDictionary<string, string> buildProperties)
+        {
+            ProjectDirectory = projectDirectory;
+            Compile = compile ?? throw new ArgumentNullException(nameof(compile));
+            ReferencePath = referencePath ?? throw new ArgumentNullException(nameof(referencePath));
+            PreprocessorSymbols = preprocessorSymbols ?? throw new ArgumentNullException(nameof(preprocessorSymbols));
+            PluginPaths = pluginPaths ?? throw new ArgumentNullException(nameof(pluginPaths));
+            IntermediateOutputDirectory = intermediateOutputDirectory ?? throw new ArgumentNullException(nameof(intermediateOutputDirectory));
+            BuildProperties = buildProperties ?? throw new ArgumentNullException(nameof(buildProperties));
+        }
 
         /// <summary>
-        /// Gets or sets the list of paths to reference assemblies.
+        /// Gets the list of paths of files to be compiled.
         /// </summary>
-        public IReadOnlyList<string> ReferencePath { get; set; }
+        public IReadOnlyList<string> Compile { get; }
 
         /// <summary>
-        /// Gets or sets a set of preprocessor symbols to define.
+        /// Gets the list of paths to reference assemblies.
         /// </summary>
-        public IEnumerable<string> PreprocessorSymbols { get; set; }
+        public IReadOnlyList<string> ReferencePath { get; }
 
         /// <summary>
-        /// Gets or sets the paths to plugins.
+        /// Gets a set of preprocessor symbols to define.
         /// </summary>
-        public IReadOnlyList<string> PluginPaths { get; set; } = new List<string>();
+        public IEnumerable<string> PreprocessorSymbols { get; }
 
         /// <summary>
-        /// Gets or sets the build properties to expose to generators.
+        /// Gets the build properties to expose to generators.
         /// </summary>
-        public IReadOnlyDictionary<string, string> BuildProperties { get; set; }
+        public IReadOnlyDictionary<string, string> BuildProperties { get; }
 
         /// <summary>
-        /// Gets or sets the path to the directory that contains generated source files.
+        /// Gets the path to the directory that contains generated source files.
         /// </summary>
-        public string IntermediateOutputDirectory { get; set; }
+        public IReadOnlyList<string> PluginPaths { get; }
```

Comment:
The comment still is wrong. It's a comment for `IntermediateOutputDirectory`. It should read
>  Gets the paths to plugins.

Reply:
Fixed myself.